app/controllers/uploaded_files_controller.rb

```diff
@@ -48,4 +60,13 @@ def show
     def uploaded_file_params
       params.require(:uploaded_file).permit(:attachment, :lecture)
     end
+
+    def validate_destroy_rights
+      @uploaded_file = UploadedFile.find(params[:id])
+      unless current_user == @uploaded_file.author
+        unless (@uploaded_file.allowsUpload.class.name == "Course") && (@uploaded_file.allowsUpload.creator_id == current_user.id)
```

Comment:
a) there is no else case so these to conditions can be merged
b) it would improve readability a lot if you store the condition result in variables with meaningfull names and then only compare the variables.

The conditions were difficult to understand for me because of the inverted logic using unless. But I think this is just a personal preference.
Anyway do I understand them correctly:
Permission is always granted if you are the author/creator of the file. And if it is a Course File and you aren't the creator permission is denied? Doesn't this result in false positives for Lecture files?

Reply:
Yes, I only added validaten for course files in this PR.
I changed it now, so it should also validate lecture files and tried to make it more readable.